File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/wowGameData/types/achievements/AchievementCategoriesModel.ts
import type { AchievementCategoryModel } from './AchievementCategoryModel';

export type AchievementCategoriesModel = {
    categories: AchievementCategoryModel[];
    rootCategories: AchievementCategoryModel[];
    guildCategories: AchievementCategoryModel[];
    totalQuantity: number;
    totalPoints: number
}

File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/WoWGameData/Character/Achievements/AchievementCategoriesModel.cs
using System.Text.Json.Serialization;

namespace CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.Achievements;

public class AchievementCategoriesModel
{
    [JsonPropertyName("categories")]
    public AchievementCategoryModel[] Categories { get; set; }

    [JsonPropertyName("root_categories")]
    public AchievementCategoryModel[] RootCategories { get; set; }

    [JsonPropertyName("guild_categories")]
    public AchievementCategoryModel[] GuildCategories { get; set; }

    public int TotalQuantity { get; set; }

    public int TotalPoints { get; set; }
}
